libcnb/examples/example-02-ruby-sample/src/layers/bundler.rs

```diff
@@ -1,34 +1,44 @@
-use std::collections::HashMap;
 use std::fs;
 use std::path::Path;
 use std::process::Command;
 
-use libcnb::data::layer_content_metadata::LayerContentMetadata;
-use libcnb::layer_lifecycle::{LayerLifecycle, ValidateResult};
+use libcnb::data::layer_content_metadata::LayerTypes;
+
 use serde::Deserialize;
 use serde::Serialize;
 use sha2::Digest;
 
 use crate::RubyBuildpack;
 use libcnb::build::BuildContext;
+use libcnb::layer::{Layer, LayerData, LayerResult, LayerResultBuilder};
+use libcnb::Env;
 
 #[derive(Deserialize, Serialize, Debug, Clone)]
 pub struct BundlerLayerMetadata {
     gemfile_lock_checksum: String,
 }
 
-pub struct BundlerLayerLifecycle {
-    pub ruby_env: HashMap<String, String>,
+pub struct BundlerLayer {
+    pub ruby_env: Env,
 }
 
-impl LayerLifecycle<RubyBuildpack, BundlerLayerMetadata, HashMap<String, String>>
-    for BundlerLayerLifecycle
-{
+impl Layer for BundlerLayer {
+    type Buildpack = RubyBuildpack;
+    type Metadata = BundlerLayerMetadata;
+
+    fn types(&self) -> LayerTypes {
```

Comment:
I agree that it would be better to enforce they stay static throughout the execution of the layer logic. However, we don't really have a good options here since traits can only contain methods.

Reply:
I presume the only option would be to have another associated type? Though that would presumably be a bit awkward for the caller.

Reply:
We're talking about values here, are you suggesting we investigate const generics for this?

Reply:
Yeah I realised that only after I replied (still waking up). I honestly have no idea. Happy to mark this as resolved and move on :-)